### PR TITLE
[Seal] Add new feature cxx14

### PR DESCRIPTION
**Pull request by WeiDaiWD**
**Describe the pull request**

- What does your PR fix? Fixes #
Added a new feature to SEAL, "cxx14", to build SEAL in C++14, which is disabled by default.

- Which triplets are supported/not supported? Have you updated the CI baseline?
Tested x64-windows-static, x64-windows-static-md, x64-linux, x64-osx.
x64-windows/x86-windows are not supported as the library only supports static linkage in Windows.
No change to the CI baseline.

- Does your PR follow the [maintainer guide](https://github.com/microsoft/vcpkg/blob/master/docs/maintainers/maintainer-guide.md)?
Yes

### CODE REVIEW

**Review by Cheney-W — CHANGES_REQUESTED** [review]
1. Please add `Port-Version: 1` after the `Version: 3.6.2` line.
2. Please run `.\vcpkg.exe x-add-version seal` to update the json file of seal.

**Review by Cheney-W — APPROVED** [review]

**Comment by Cheney-W** [comment]
@WeiDaiWD 
Thanks for your so quick response. About port's json file, each of your commit changes will modify its `git tree` value, which means you need to update json file after you have done all the changes and executed the git commit -m "XXX" command. The specific sequence is as follows:
```
Git checkout -b "xxx"
Modify the local files
Git diff
Git add .
Git commit -m "comments description"
.\vcpkg x-add-version {Port name}
Git add .
Git commit -m "comments description"
git push
```
If you haven't executed the '. \vcpkg x-add-version {port name}' on the current branch, you need to use it after your modification is completed and submitted.
If you have executed this command on the current branch, you need to execute the following command when you need to update the json file:
`.\vcpkg.exe x-add-version {port name} --overwrite-version`

So, for this PR, you need to execute `.\vcpkg.exe x-add-version {port name} --overwrite-version` again.

**Comment by WeiDaiWD** [comment]
Thanks for the help! Hope I do it correctly this time.

**Comment by Cheney-W** [comment]
osgearth:x64-linux failure is not related with your PR, this is a CI issue, we are working on it.

File: ports/seal/CONTROL [hunk 1]
```diff
@@ -20,3 +21,7 @@ Description: Use zstd for compressed serialization
 Feature: no-throw-tran
 Build-Depends:
 Description: Do not throw when operating on transparent ciphertexts
+
+Feature: cxx14
+Build-Depends:
+Description: Uses C++14 instead of 17
```
Comment (PhoebeHui) [inline ports/seal/CONTROL:27]: This breaks our policy, we would not accept today,  see https://github.com/microsoft/vcpkg/blob/master/docs/maintainers/maintainer-guide.md#do-not-use-features-to-control-alternatives-in-published-interfaces

Comment (WeiDaiWD) [reply ports/seal/CONTROL:27]: Got it. I would recommend users in favor of C++14 to build SEAL without vcpkg. I'm closing this PR now. Thanks for the help!

Comment (PhoebeHui) [reply ports/seal/CONTROL:27]: You can also suggest user to build it with overlay ports instead.

**Pull request closed without merging**